File: api/routers/domain.py
from fastapi import APIRouter, HTTPException
from ..services.dns_checker import check_domain_health

router = APIRouter(prefix="/domain", tags=["Domain"])

@router.get("/health/{domain}")
async def domain_health(domain: str):
    domain = domain.lower().strip().lstrip("https://").lstrip("http://").lstrip("www.").rstrip("/")
    if not domain:
        raise HTTPException(422, "Invalid domain.")
    return await check_domain_health(domain)

File: api/routers/domain.js
import { Router } from "express";
import { checkDomainHealth } from "../services/dnsChecker.js";

export const domainRouter = Router();

domainRouter.get("/domain/health/:domain", async (req, res) => {
  let domain = req.params.domain.toLowerCase()
    .replace(/^https?:\/\//, "").replace(/^www\./, "").replace(/\/$/, "");
  if (!domain) return res.status(422).json({ error: "Invalid domain." });
  try {
    const result = await checkDomainHealth(domain);
    res.json(result);
  } catch (e) {
    res.status(500).json({ error: e.message });
  }
});
